vars/runATH.groovy

```diff
@@ -0,0 +1,156 @@
+        } else if (jenkins == "latest-rc") {
+            jenkinsURl = mirror + "/war-rc/latest/jenkins.war"
+        } else if (jenkins == "lts") {
+            jenkinsURl = mirror + "war-stable/latest/jenkins.war"
+        } else if (jenkins == "lts-rc") {
+            jenkinsURl = mirror + "war-stable-rc/latest/jenkins.war"
+        }
+
+        def canFindJenkinsWar = sh(script: "curl --output /dev/null --silent --fail -r 0-0 -L '${jenkinsURl}'", returnStatus: true) == 0
+        if (!canFindJenkinsWar) {
+            error "Is not possible to find the given jenkins war file, please review it. Current value for jenkins is ${jenkins}"
+        }
+        // Validation ended
+
+        // ATH
+        if (isLocalATH) { // Deal with already existing ATH sources
+            athSourcesFolder = athUrl - "file://"
+        } else {
+            dir(athSourcesFolder) {
+                checkout changelog: true, poll: false, scm: [$class           : 'GitSCM', branches: [[name: "${athRevision}"]],
+                                                             userRemoteConfigs: [[url: "${athUrl}"]]]
+            }
+        }
+        dir(athSourcesFolder) {
+            def uid = sh(script: "id -u", returnStdout: true)
+            def gid = sh(script: "id -g", returnStdout: true)
+            athContainerImage = docker.build('jenkins/ath', "--build-arg=uid='$uid' --build-arg=gid='$gid' -f src/main/resources/ath-container/Dockerfile .")
+            // We may need to run things in parallel later, so avoid several checkouts
+            stash name: "athSources"
+        }
+
+        // Jenkins war
+        sh("curl -o jenkins.war -L ${jenkinsURl}")
+        stash includes: '*.war', name: 'jenkinsWar'
+
+    }
+    // Workaround for https://issues.jenkins-ci.org/browse/JENKINS-27092
+    if (shouldStop) {
+
+        return
+    }
+    // End of workaround
+
+    stage("Running ATH") {
+        def testsToRun = metadata.tests?.join(",")
+        def categoriesToRun = metadata.categories?.join(",")
+        def browsers = metadata.browsers ?: ["firefox"]
+        def failFast = metadata.failFast ?: false
+        def rerunCount = metadata.rerunFailingTestsCount ?: 0
+        def localSnapshots = metadata.useLocalSnapshots ?: true
+
+        if (testsToRun == null && categoriesToRun == null) {
+            categoriesToRun = defaultCategory
+        }
+
+        def testingbranches = ["failFast": failFast]
+        for (browser in browsers) {
+            // TODO remove this if else block once other browser are supported by infra
+            if (browser == "firefox") {
+
+                def currentBrowser = browser
+                def containerArgs = "-v /var/run/docker.sock:/var/run/docker.sock -e LOCAL_SNAPSHOTS=${localSnapshots} -e SHARED_DOCKER_SERVICE=true -u ath-user"
+                def commandBase = "./run.sh ${currentBrowser} jenkins.war -Dmaven.test.failure.ignore=true -DforkCount=1 -B -Dsurefire.rerunFailingTestsCount=${rerunCount}"
```

Comment:
BTW the `{}` are not necessary for most cases of string interpolation in Groovy.

```groovy
def commandBase = "./run.sh $currentBrowser jenkins.war -Dmaven.test.failure.ignore=true -DforkCount=1 -B -Dsurefire.rerunFailingTestsCount=$rerunCount"
```

should work the same.

Reply:
I find easier to read code when {} is explicitly used even if not needed. I can change that if there is a coding standard or technical reason, but I prefer to let it be like is if not.

Unless there are more people demanding the change...

Reply:
Nope, just FYI.